From b303a4b87b8478c1519aed14ccdbebe1f09e64da Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Tue, 27 Oct 2015 20:50:48 +0100
Subject: [PATCH 2/2] New configure option --disable-delta-protocol

A third party project without a working delta protocol implementation may
need the ability to communicate with Freeciv. Today this must be done by
patching generate_packets.py.

Let the build system disable the delta network protocol. Emit a warning if
it is disabled.

Turning off bool folding when the delta protocol otherwise is supported is
still not possible.
---
 common/Makefile.am         |  2 +-
 common/generate_packets.py |  8 +++++---
 configure.ac               | 17 +++++++++++++++++
 3 files changed, 23 insertions(+), 4 deletions(-)

diff --git a/common/Makefile.am b/common/Makefile.am
index 984cee5..15907ec 100644
--- a/common/Makefile.am
+++ b/common/Makefile.am
@@ -159,7 +159,7 @@ BUILT_SOURCES = packets_gen.c packets_gen.h
 packets_gen.h packets_gen.c: packets_generate
 .INTERMEDIATE: packets_generate
 packets_generate: packets.def generate_packets.py
-	cd $(srcdir) && ./generate_packets.py
+	cd $(srcdir) && ./generate_packets.py ${GENERATE_PACKETS_ARGS}
 	touch packets_generate
 
 # These files are not generated to builddir, but to srcdir */
diff --git a/common/generate_packets.py b/common/generate_packets.py
index cc1731f..545979e 100755
--- a/common/generate_packets.py
+++ b/common/generate_packets.py
@@ -27,7 +27,6 @@ generate_variant_logs=1
 
 ### The following parameters CHANGE the protocol. You have been warned.
 fold_bool_into_header=1
-disable_delta=0
 
 ################# END OF PARAMETERS ####################
 
@@ -39,6 +38,9 @@ import re, string, os, sys
 
 lazy_overwrite=0
 
+def is_delta_disabled():
+    return "--disable-delta" in sys.argv
+
 def verbose(s):
     if "-v" in sys.argv:
         print(s)
@@ -169,7 +171,7 @@ def parse_fields(str, types):
     if flaginfo["is_key"]: arr.remove("key")
     flaginfo["diff"]=("diff" in arr)
     if flaginfo["diff"]: arr.remove("diff")
-    if disable_delta:
+    if is_delta_disabled():
         flaginfo["diff"] = 0
     adds=[]
     removes=[]
@@ -1340,7 +1342,7 @@ class Packet:
 
         assert len(arr)==0,repr(arr)
 
-        if disable_delta:
+        if is_delta_disabled():
             self.delta=0
 
         self.fields=[]
diff --git a/configure.ac b/configure.ac
index b5d8d63..8790277 100644
--- a/configure.ac
+++ b/configure.ac
@@ -268,6 +268,23 @@ dnl checks for MagickWand mapimg support
 dnl sets MAPIMG_WAND_CFLAGS, MAPIMG_WAND_LIBS
 FC_MAPIMG_MAGICKWAND
 
+dnl set arguments for the packet generator
+GENERATE_PACKETS_ARGS=""
+
+dnl make it possible to disable the delta network protocol
+AC_ARG_ENABLE([delta-protocol],
+  AS_HELP_STRING([--disable-delta-protocol],
+                 [disable the delta network protocol]))
+
+AS_IF([test "x$enable_delta_protocol" = "xno"], [
+  AC_MSG_WARN([delta network protocol is off])
+  AC_MSG_WARN([no network compatibility with regular Freeciv])
+  AC_MSG_WARN([won't be able to play with regular Freeciv])
+  GENERATE_PACKETS_ARGS="$GENERATE_PACKETS_ARGS --disable-delta"])
+
+dnl done setting arguments for the packet generator
+AC_SUBST([GENERATE_PACKETS_ARGS])
+
 FC_WEB_OPTIONS
 
 AC_ARG_ENABLE([fcweb],
-- 
2.1.4

